lib/icinga/dependency-group.cpp

```diff
@@ -0,0 +1,362 @@
+		m_Members.emplace(compositeKey, MemberValueType{{member->GetChild().get(), member.get()}});
+	}
+}
+
+/**
+ * Remove a member from the current dependency group.
+ *
+ * @param member The dependency to remove from the dependency group.
+ */
+void DependencyGroup::RemoveMember(const Dependency::Ptr& member)
+{
+	std::lock_guard lock(m_Mutex);
+	if (auto it(m_Members.find(MakeCompositeKeyFor(member))); it != m_Members.end()) {
+		auto [begin, end] = it->second.equal_range(member->GetChild().get());
+		for (auto memberIt(begin); memberIt != end; ++memberIt) {
+			if (memberIt->second == member) {
+				// This will also remove the child Checkable from the multimap container
+				// entirely if this was the last member of it.
+				it->second.erase(memberIt);
+				// If the composite key has no more members left, we can remove it entirely as well.
+				if (it->second.empty()) {
+					m_Members.erase(it);
+					m_CompositeKey = ""; // Invalidate the composite key cache (if any).
+				}
+				return;
+			}
+		}
+	}
+}
+
+/**
+ * Move the members of the provided dependency group to the provided destination dependency group.
+ *
+ * @param dest The dependency group to move the members to.
+ */
+void DependencyGroup::MoveMembersTo(const DependencyGroup::Ptr& dest)
+{
+	VERIFY(this != dest); // Prevent from doing something stupid, i.e. deadlocking ourselves.
+
+	std::lock_guard lock(m_Mutex);
+	for (auto& [_, members] : m_Members) {
+		std::for_each(members.begin(), members.end(), [&dest](const auto& member) {
+			dest->AddMember(member.second);
+		});
+	}
+}
+
+/**
+ * Set the Icinga DB identifier for the current dependency group.
+ *
+ * The only usage of this function is the Icinga DB feature used to cache the unique hash of this dependency groups.
+ *
+ * @param identifier The Icinga DB identifier to set.
+ */
+void DependencyGroup::SetIcingaDBIdentifier(const String& identifier)
+{
+	std::lock_guard lock(m_Mutex);
+	m_IcingaDBIdentifier = identifier;
+}
+
+/**
+ * Retrieve the Icinga DB identifier for the current dependency group.
+ *
+ * When the identifier is not already set by Icinga DB via the SetIcingaDBIdentifier method,
+ * this will just return an empty string.
+ *
+ * @return - Returns the Icinga DB identifier for the current dependency group.
+ */
+String DependencyGroup::GetIcingaDBIdentifier() const
+{
+	std::lock_guard lock(m_Mutex);
+	return m_IcingaDBIdentifier;
+}
+
+/**
+ * Retrieve the (non-unique) name of the current dependency group.
+ *
+ * For explicitly configured redundancy groups, the name of the dependency group is the same as the one located
+ * in the configuration files. For non-redundant dependencies, on the other hand, the name is always empty.
+ *
+ * @return - Returns the name of the current dependency group.
+ */
+const String& DependencyGroup::GetName() const
+{
+	// We don't need to lock the mutex here, as the name is set once during
+	// the object construction and never changed afterwards.
+	return m_Name;
+}
```

Comment:
Again, to make things more explicit, I'd rename this to `GetReundancyGroupName()`. Should also be fine documentation-wise if it then just says "If this does not represent a redundancy group, an empty string is returned".